colossalai/inference/modeling/models/nopadding_bloom.py

```diff
@@ -0,0 +1,806 @@
+        fd_inter_tensor (FDIntermTensors): Holding tensors used for
+            storing intermediate values in flash-decoding.
+        is_prompts (bool, optional): Whether the current inference process is in the context input phase. Defaults to True.
+        kv_seq_len (int, optional): The max sequence length of input sequences. Defaults to 0.
+        output_tensor (torch.Tensor, optional): The mid tensor holds the output of attention. Defaults to None.
+        norm_output (torch.Tensor, optional): The mid tensor holds the output of layernorm. Defaults to None.
+        sm_scale (int, optional): Used for flash attention. Defaults to None.
+        use_cuda_kernel: (bool, optional): Whether to use cuda kernel. Defaults to True.
+        cu_seqlens(torch.Tensor, optional): Holding the cumulative sum of sequence length.
+        high_precision(Optional[bool]): Whether to use float32 for underlying calculations of float16 data to achieve higher precision, defaults to False.
+
+    Returns:
+        torch.Tensor: The output tensor.
+    """
+
+    # print(f"[BloomBlock] input hidden_states {hidden_states}")
+
+    # LayerNorm before attention
+    norm_output = self.input_layernorm(hidden_states)
+
+    if self.apply_residual_connection_post_layernorm:
+        residual = norm_output
+    else:
+        residual = hidden_states
+
+    # Self attention
+    attn_outputs = self.self_attention(
+        hidden_states=norm_output,
+        block_tables=block_tables,
+        k_cache=k_cache,
+        v_cache=v_cache,
+        is_prompts=is_prompts,
+        sequence_lengths=sequence_lengths,
+        fd_inter_tensor=fd_inter_tensor,
+        kv_seq_len=kv_seq_len,
+        output_tensor=output_tensor,
+        sm_scale=sm_scale,
+        use_cuda_kernel=use_cuda_kernel,
+        cu_seqlens=cu_seqlens,
+        high_precision=high_precision,
+    )
+
+    # attention_output = attn_outputs[0]
+    # outputs = attn_outputs[1:]
+    attention_output = attn_outputs + residual
+
+    # LayerNorm post attention
+    norm_output = self.post_attention_layernorm(attention_output)
+
+    if self.apply_residual_connection_post_layernorm:
+        residual = norm_output
+    else:
+        residual = attention_output
+
+    # MLP (including residuals)
+    output = self.mlp(norm_output, residual)
+
+    # print(f"[DEBUG] output shape {output.shape}, and outputs shape {outputs.shape}")
+    # print(f"[DEBUG] output type {output.dtype}, and outputs type {outputs.dtype}")
+    # outputs = output + outputs
+
+    # return outputs
+
+    # print(f"[BloomBlock] output {output}")
+    return output
+
+
+# class NopadBloomAttention(nn.Module):
+#     def __init__(
+#         self,
+#         hidden_size: int,
+#         n_heads: int,
+#         attn_qproj_w: torch.Tensor = None,
+#         attn_kproj_w: torch.Tensor = None,
+#         attn_vproj_w: torch.Tensor = None,
+#         attn_oproj_w: torch.Tensor = None,
+#     ):
```

Comment:
As the end2end flow for bloom is ready and none of the attention and mlp classes is going to be used, you might want to remove these classes.